client/src/components/Tabs/BasicTabs.js

```diff
@@ -0,0 +1,31 @@
+import React, { useEffect, useState } from "react";
+import { Tabs } from "antd";
+
+const { TabPane } = Tabs;
+
+const BasicTabs = ({ callback, tabs, position, def, disabled }) => {
+  useEffect(() => {
+    console.log(tabs);
+  });
+
+  return (
+    <Tabs
+      tabPosition={position ? position : "top"}
+      defaultActiveKey={def ? def : "0"}
+      onChange={callback}
+    >
+      {tabs.map((e, index) => {
+        if (disabled) {
+          if (disabled[index] === true) {
+            return <TabPane tab={e} disabled key={index}></TabPane>;
```

Comment:
The view that will be displayed each tab is the children of the <TabPane> components. The current props does not allow the parent component of this basic tab to pass in the children of the TabPane, so it's not possible to render anything other than an empty view with the current design right now. For example in this demo code:
const Demo = () => (
  <Tabs defaultActiveKey="1" onChange={callback}>
    <TabPane tab="Tab 1" key="1">
      Content of Tab Pane 1
    </TabPane>
    <TabPane tab="Tab 2" key="2">
      Content of Tab Pane 2
    </TabPane>
    <TabPane tab="Tab 3" key="3">
      Content of Tab Pane 3
    </TabPane>
  </Tabs>
);
The "Content of Tab Pane 1", Content of Tab Pane 2" and Content of Tab Pane 3" are the children of the TabPane, and we need a way to pass such children into these pane.

Reply:
That is the traditional way these tabs are used to display/hide content is through the child/parent hierarchy of the tabPanes. To make it more reusable in other places, and to avoid a bunch of div nesting we are using a more stateful approach. Let me elaborate

You would setup a piece of state in your component, let's call it view

`const [view, setView] = useState(pane1);`

Every time you click a different Tab the callback you pass into BasicTab changes your state so click TabPane1 the view is pane1, click TabPane2 and view=pane2 etc.

this allows you to organize your components like this
```
<BasicTabs />
{pane1 ? Content of Tab Pane 1 : null }
{pane2 ? Content of Tab Pane 2 : null }
{pane3 ? Content of Tab Pane 3 : null }

```

Otherwise anytime anyone wants to create tabs, they will have to just make the tabs from scratch inside of their components which isn't super heard, just a pretty W.E.T way of doing it.

Does that make sense? Maybe we need to meet to walk through it?

Reply:
I'm a bit concern since the way you implement it may break the expectation of antd-Tab component. The antd-Tab component expects the view of each Tab to be the children of TabPane. It does all the animation and views switching and view placement logic there. If you move the tab view to be the sibling of the parent <Tab> component instead, then the consumer of your tab component has to re-implement view switching logic, animation, and placement themselves (for example, using the react hook like in your example for view switching), and that is a lot of core logic to be re-implemented. Can you upload a video of this tab switching between different views with some placeholder text in it? I think if you try that you'll see the animation that comes with the Tab component will be missing, and you have to manually place the view in the correct location via flexbox.
To address your concern regarding the reusability of BasicTab, it can be achieved without compromising the built-in functionality of the Tab component bypassing the children's view itself as props to the Basic Tab component. For example, if you look at `AvatarModal` component in the code base, You'll see the footer prop is a React component that is being passed through. We can achieve reusability using this pattern, so instead of just passing to BasicTab the tab name, it can pass an object which includes {'tabName': ..., 'tabView': [a React component], disabled: [boolean]} This can also help remove the disabled array and wrap it in the same object. I believe doing this way will allow us to reuse more functionality from the tab component without compromising the reusability of the basic tab component that you build.

Reply:
The functionality is the same regardless (See below with the two different structures compared). It basically comes down to if you want to write your components inplace with all the rest of your components, or if you want to write them above in a view object which is passed in as a prop. If the latter makes more sense to you, it'll probably make more sense to everyone else. I'll go ahead a refactor

Nesting the components
![FightPandemics](https://user-images.githubusercontent.com/5578094/108111884-32f8f780-704a-11eb-9705-4003bd6e4413.gif)

Controlling with state
![FP - Stateful tabs](https://user-images.githubusercontent.com/5578094/108112018-62a7ff80-704a-11eb-92ac-d290372a4753.gif)